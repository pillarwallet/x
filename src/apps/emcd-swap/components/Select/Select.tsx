import React, { useState, useRef, useEffect } from "react";

import SelectActivator from './components/SelectActivator';
import SelectDropdown from './components/SelectDropdown';

export interface Option {
   [key: string]: any;
   id?: string | number;
   name?: string;
   icon_url?: string;
}
interface CustomSelectProps {
  options: Option[];
  currentValue: Record<string, any> | null;
  placeholder: string;
  itemValue?: string;
  itemText?: string; // Field for displaying text (default is "name")
  withIcon?: boolean;
  itemIcon?: string;
  className?: string; // Allow additional classes for styling
  onChange?: (value: Option) => void; // Callback when an option is selected
}

const CustomSelect: React.FC<CustomSelectProps> = ({
  options,
  placeholder,
  currentValue,
  itemValue = "id",
  itemText = "name",
  onChange,
  withIcon,
  className,
  itemIcon,
}) => {
  const [isOpen, setIsOpen] = useState(false);
  const [selected, setSelected] = useState<Option | null>(null);
  const [highlightedIndex, setHighlightedIndex] = useState(0);
  const dropdownRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (currentValue) {
      setSelected(currentValue);
      const currentIndex = options.findIndex(
        (opt) => opt[itemValue] === currentValue[itemValue]
      );
      if (currentIndex !== -1) setHighlightedIndex(currentIndex);
    } else {
      setSelected(null)
      setHighlightedIndex(0);
    }
  }, [currentValue, options, itemValue]);

  const toggleDropdown = () => setIsOpen((prev) => !prev);

  const handleOptionClick = (option: Option) => {
    setSelected(option);
    onChange?.(option);
    setIsOpen(false);
  };

  const handleClickOutside = (event: MouseEvent) => {
    if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
      setIsOpen(false);
    }
  };

  useEffect(() => {
    document.addEventListener("mousedown", handleClickOutside);
    return () => {
      document.removeEventListener("mousedown", handleClickOutside);
    };
  }, []);

  const handleKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
    if (e.key === "Escape" && isOpen) {
      setIsOpen(false);
    } else if (e.key === "Enter" || e.key === " ") {
      if (isOpen) {
        if (options.length > 0) {
          handleOptionClick(options[highlightedIndex]);
        }
      } else {
        setIsOpen(true);
      }
      e.preventDefault(); // Prevent page scroll for Space key
    } else if (e.key === "ArrowDown" && isOpen) {
      if (options.length > 0) {
        setHighlightedIndex((prev) => (prev + 1) % options.length);
      }
      e.preventDefault();
    } else if (e.key === "ArrowUp" && isOpen) {
      if (options.length > 0) {
        setHighlightedIndex((prev) => (prev - 1 + options.length) % options.length);
      }
      e.preventDefault();
    }
  };

  return (
    <div
      ref={dropdownRef}
      className={`relative w-36 ${className || ''}`}
      tabIndex={0} // to allow div to receive focus
      onKeyDown={handleKeyDown}
      role="combobox"
      aria-haspopup="listbox"
      aria-expanded={isOpen}
      aria-controls="select-listbox"
      aria-activedescendant={
        isOpen && options.length > 0 && options[highlightedIndex]
          ? `option-${options[highlightedIndex][itemValue]}`
          : undefined
      }
    >
      <SelectActivator
        isOpen={isOpen}
        option={selected}
        placeholder={placeholder}
        itemText={itemText}
        itemIcon={itemIcon}
        withIcon={withIcon}
        toggleDropdown={toggleDropdown}
      />

      {isOpen && (
        <SelectDropdown
          id="select-listbox"
          options={options}
          itemValue={itemValue}
          itemText={itemText}
          withIcon={withIcon}
          itemIcon={itemIcon}
          handleOptionClick={handleOptionClick}
          highlightedIndex={highlightedIndex}
        />
      )}
    </div>
  );
};

export default CustomSelect;