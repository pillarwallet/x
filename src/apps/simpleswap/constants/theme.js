export const themes = {
  white: {
    value: 'white',
    background: '#FFFFFF',
    background1: '#ECF1F7',
    background1hover: '#E4EBF3',
    background2: '#C6D5EA4D',
    arrowBlockBackground: '#ECF1F7',
    dropdownBackground: '#C6D5EA80',
    dropdownHoverBackground: '#C6D5EAB3',
    buttonBackground: '#0F75FC',
    buttonHoverBackground: '#004AD9',
    checkboxColor: '#FFFFFF',
    checkboxHoverBorder: '#3F5878',
    goBackBackground: 'rgba(198, 213, 234, 0.3)',
    copyIconBackground: '#C6D5EA4D',
    copyIconHoverBackground: '#2D3D5D',
    addressBlockBackground: '#ECF1F7',
    addressBlockText: '#859AB5',
    currentStep: '#3F5878',
    allSteps: '#859AB5',
    text1: '#252C44',
    text2: '#3F5878',
    text3: '#859AB5',
    addressInfoText: '#859AB5',
    messageTextPlaceholder: '#859AB5',
    border: '#ECF1F7',
    inputActiveBorder: '#859AB5',
    dropdownCoinName: '#859AB5',
    dropdownSectionTitle: '#3F5878',
    exchangeIdValue: '#859AB5',
    rateTextHover: '#3F5878',
    hoverCloseIcon: '#3F5878',
    closeIcon: '#859AB5',
    searchIcon: '#859AB5',
    copyIcon: '#859AB5',
    addressInfoCopyIcon: '#859AB5',
    textLink: '#0F75FC',
    textLinkHover: '#004AD9',
    buttonText: '#FFFFFF',
    warningCopyIconBackground: '#FEF7EA',
    warningCopyIconHover: '#FFFFFF',
    tooltipText: '#FFF',
    activeIconStroke: '#C6D5EA',
  },
  black: {
    value: 'black',
    background: '#1B2B4B',
    background1: '#2D3D5D',
    background1hover: '#3F5878',
    background2: '#2D3D5D80',
    arrowBlockBackground: '#3F5878',
    dropdownBackground: '#2D3D5D80',
    dropdownHoverBackground: '#3F587880',
    buttonBackground: '#0F75FC',
    buttonHoverBackground: '#004AD9',
    checkboxColor: '#FFFFFF',
    checkboxHoverBorder: '#C6D5EA',
    goBackBackground: 'rgba(198, 213, 234, 0.3)',
    copyIconBackground: '#C6D5EA4D',
    copyIconHoverBackground: '#2D3D5D',
    addressBlockBackground: '#ECF1F7',
    addressBlockText: '#859AB5',
    currentStep: '#859AB5',
    allSteps: '#3F5878',
    text1: '#FFFFFF',
    text2: '#859AB5',
    text3: '#859AB5',
    addressInfoText: '#C6D5EA',
    messageTextPlaceholder: '#859AB5',
    border: '#3F5878',
    inputActiveBorder: '#859AB5',
    dropdownCoinName: '#859AB5',
    dropdownSectionTitle: '#3F5878',
    exchangeIdValue: '#C6D5EA',
    rateTextHover: '#C6D5EA',
    hoverCloseIcon: '#C6D5EA',
    closeIcon: '#859AB5',
    searchIcon: '#859AB5',
    copyIcon: '#859AB5',
    addressInfoCopyIcon: '#859AB5',
    textLink: '#0F75FC',
    textLinkHover: '#004AD9',
    buttonText: '#FFFFFF',
    warningCopyIconBackground: '#EE950033',
    warningCopyIconHover: '#FFFFFF',
    tooltipText: '#252C44',
    activeIconStroke: '#4D678A',
  },
  blue: {
    value: 'blue',
    background: '#0F75FC',
    background1: '#0A64F3',
    background1hover: '#0C61E9',
    background2: '#004AD980',
    arrowBlockBackground: '#7CB2F9',
    dropdownBackground: '#004AD980',
    dropdownHoverBackground: '#004AD9B3',
    buttonBackground: '#FFFFFF',
    buttonHoverBackground: '#004AD9',
    swapRowHoverBackground: '#004AD9',
    swapIconColor: '#FFFFFF',
    checkboxColor: '#0F75FC',
    checkboxHoverBorder: '#F2F7FE',
    goBackBackground: '#7CB2F9',
    copyIconBackground: '#F2F7FE',
    copyIconHoverBackground: '#004AD9',
    addressBlockBackground: '#FFFFFF',
    addressBlockText: '#7CB2F9',
    currentStep: '#FFFFFF',
    allSteps: '#7CB2F9',
    text1: '#FFFFFF',
    text2: '#B9D7FE',
    text3: '#B9D7FE',
    addressInfoText: '#F2F7FE',
    messageTextPlaceholder: '#7CB2F9',
    border: '#7CB2F9',
    inputActiveBorder: '#C6D5EA',
    dropdownCoinName: '#F2F7FE',
    dropdownSectionTitle: '#7CB2F9',
    exchangeIdValue: '#F2F7FE',
    rateTextHover: '#B9D7FE',
    hoverCloseIcon: '#004AD9',
    closeIcon: '#7CB2F9',
    searchIcon: '#F2F7FE',
    copyIcon: '#7CB2F9',
    addressInfoCopyIcon: '#F2F7FE',
    textLink: '#FFFFFF',
    textLinkHover: '#FFFFFF',
    buttonText: '#0F75FC',
    warningCopyIconBackground: '#FEF7EA',
    warningCopyIconHover: '#EE9500',
    tooltipText: '#0F75FC',
    activeIconStroke: '#7CB2F9',
  },
};
