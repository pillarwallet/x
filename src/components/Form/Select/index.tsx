/* eslint-disable @typescript-eslint/no-use-before-define */
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { PiCaretDown, PiCaretUp } from 'react-icons/pi';
import styled from 'styled-components';

// types
import { SelectOption } from '../../../types';

// components
import RandomAvatar from '../../../apps/pillarx-app/components/RandomAvatar/RandomAvatar';
import { CompatibleChains } from '../../../utils/blockchain';
import SkeletonLoader from '../../SkeletonLoader';
import Alert from '../../Text/Alert';

const ListItem = ({
  option,
  onClick,
  rightAddon,
  hideValue,
}: {
  option: SelectOption;
  onClick?: (option: SelectOption) => void;
  rightAddon?: React.ReactNode;
  hideValue?: boolean;
}) => {
  const [hideImage, setHideImage] = useState(false);

  const isChainName = CompatibleChains.map((chains) => chains.chainName).some(
    (chain) => option.title.includes(chain)
  );

  return (
    <ListItemWrapper
      onClick={() => onClick && onClick(option)}
      id={`action-bar-send-${option.title}`}
    >
      {option.imageSrc && !hideImage && (
        <ListItemLeft>
          <ListItemImage
            src={option.imageSrc}
            alt={option.title}
            title={option.title}
            onError={() => setHideImage(true)}
          />
        </ListItemLeft>
      )}
      {!option.imageSrc && !isChainName && (
        <ListItemLeft>
          <RandomAvatar name={option.title} isRound variant="marble" />
        </ListItemLeft>
      )}
      <ListItemRight>
        <ListItemTitle>{option.title}</ListItemTitle>
        {!hideValue && (
          <>
            {!option.isLoadingValue && (
              <ListItemValue>{option.value}</ListItemValue>
            )}
            {option.isLoadingValue && (
              <SkeletonLoader
                data-testid="skeleton-loader-list-item"
                $height="15px"
                $width="30%"
              />
            )}
          </>
        )}
      </ListItemRight>
      {rightAddon}
    </ListItemWrapper>
  );
};

export const Select = ({
  options,
  defaultSelectedId,
  onChange,
  isLoadingOptions,
  hideValue,
  type,
}: {
  options: SelectOption[];
  defaultSelectedId?: string;
  onChange: (option: SelectOption) => void;
  isLoadingOptions?: boolean;
  hideValue?: boolean;
  type?: 'token' | 'nft';
}) => {
  const [t] = useTranslation();
  const [expanded, setExpanded] = useState(false);
  const [selectedId, setSelectedId] = useState(defaultSelectedId);

  if (isLoadingOptions) {
    return (
      <ListItemWrapper>
        <ListItemLeft>
          <SkeletonLoader $height="42px" $width="42px" $radius="50%" />
        </ListItemLeft>
        <ListItemRight>
          <SkeletonLoader $height="15px" $width="80%" />
          {!hideValue && <SkeletonLoader $height="15px" $width="30%" />}
        </ListItemRight>
      </ListItemWrapper>
    );
  }

  const selected = options.find((option) => option.id === selectedId);

  return (
    <div>
      {selected && (
        <ListItem
          option={selected}
          rightAddon={
            <SelectToggleButton>
              {expanded ? <PiCaretUp /> : <PiCaretDown />}
            </SelectToggleButton>
          }
          onClick={() => setExpanded(!expanded)}
          hideValue={hideValue}
        />
      )}
      {(expanded || !selected) &&
        options
          .filter((option) => option.id !== selected?.id)
          .map((option) => (
            <ListItem
              key={option.id}
              option={option}
              onClick={() => {
                setExpanded(false);
                setSelectedId(option.id);
                onChange(option);
              }}
              hideValue={hideValue}
            />
          ))}
      {!options.length && (
        <Alert>
          {type === 'token' ? t`error.noTokensFound` : t`error.noNftsFound`}
        </Alert>
      )}
    </div>
  );
};

const ListItemImage = styled.img`
  width: 42px;
  height: 42px;
  border-radius: 50%;
`;

const SelectToggleButton = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding-right: 5px;
`;

const ListItemLeft = styled.div`
  width: 42px;
`;

const ListItemRight = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
`;

const ListItemTitle = styled.p`
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
`;

const ListItemValue = styled.p`
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
`;

const ListItemWrapper = styled.div`
  border-radius: 10px;
  background: ${({ theme }) => theme.color.background.selectItem};
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;

  ${({ onClick, theme }) =>
    onClick &&
    `
    &:hover {
      background: ${theme.color.background.selectItemHover};
    }
  `}
`;

export default Select;
