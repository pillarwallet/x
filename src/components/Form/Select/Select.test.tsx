import { act, render, RenderResult } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';

// components
import Select, { SelectOption } from '.';

// theme
import { defaultTheme } from '../../../theme';

const testOptions: SelectOption[] = [
  {
    id: '1',
    title: 'test 1',
    value: 'test1',
  },
  {
    id: '2',
    title: 'test 2',
    value: 'test2',
  },
  {
    id: '3',
    title: 'test 3',
    value: 'test3',
    imageSrc: 'https://via.placeholder.com/150',
  },
];

describe('<Select />', () => {
  let rendered: RenderResult;

  it('renders correctly with no options', async () => {
    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select options={[]} onChange={() => {}} />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();
  });

  it('renders correctly with some options', async () => {
    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select options={testOptions} onChange={() => {}} />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();

    const select = rendered.container.children?.item(0) as Element;
    expect(select.children?.length).toBe(testOptions.length); // none selected, all list shown
  });

  it('responds once clicked', async () => {
    const user = userEvent.setup();

    const onChange = jest.fn();

    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select options={testOptions} onChange={onChange} />
        </ThemeProvider>
      );
    });

    const select = rendered.container.children?.item(0) as Element;
    const option2 = select.children?.item(1) as Element;
    await user.click(option2);

    expect(rendered.asFragment()).toMatchSnapshot();

    expect(select.children?.length).toBe(1); // only selected shown
    expect(onChange).toBeCalledWith(expect.objectContaining(testOptions[1]));
  });

  it('renders correctly with default option', async () => {
    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select
            options={testOptions}
            onChange={() => {}}
            defaultSelectedId={testOptions[1].id}
          />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();

    const select = rendered.container.children?.item(0) as Element;
    expect(select.children?.length).toBe(1); // only selected shown

    const secondOption = select.children?.item(0) as Element;
    const secondOptionContent = secondOption.children?.item(0) as Element;
    const secondOptionValue = secondOptionContent.children?.item(1) as Element;
    expect(secondOptionValue.innerHTML).toBe(testOptions[1].value);
  });

  it('shows loading skeleton when options loading', async () => {
    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select options={testOptions} isLoadingOptions onChange={() => {}} />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();

    const select = rendered.container.children?.item(0) as Element;
    expect(select.children?.length).toBe(2); // loading skeleton left&right components shown

    const animationLeft = select.children.item(0);
    const animationRight = select.children.item(1);
    const animationMatch = expect.stringContaining('1s linear infinite');
    expect(animationLeft?.children.item(0)).toHaveStyleRule(
      'animation',
      animationMatch
    );
    expect(animationRight?.children.item(0)).toHaveStyleRule(
      'animation',
      animationMatch
    );
  });

  it('shows loading skeleton on option item where option value is loading', async () => {
    const testOptionsWithLoadingValue = testOptions.map((option, index) => ({
      ...option,
      isLoadingValue: index === 1, // 2nd option value is loading
    }));

    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select options={testOptionsWithLoadingValue} onChange={() => {}} />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();

    const select = rendered.container.children?.item(0) as Element;
    expect(select.children?.length).toBe(3); // all options shown as usual

    const secondOption = select.children?.item(1) as Element;
    const secondOptionContent = secondOption.children?.item(0) as Element;
    const secondOptionValue = secondOptionContent.children?.item(1) as Element;
    const animationMatch = expect.stringContaining('1s linear infinite');
    expect(secondOptionValue).toHaveStyleRule('animation', animationMatch);
  });

  it('renders correctly with option image', async () => {
    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select options={testOptions} onChange={() => {}} />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();

    const select = rendered.container.children?.item(0) as Element;
    const thirdOption = select.children?.item(2) as Element;
    const thirdOptionLeft = thirdOption.children?.item(0) as Element;
    const thirdOptionImage = thirdOptionLeft.children?.item(0) as Element;
    expect(thirdOptionImage).toHaveAttribute('src', testOptions[2].imageSrc);
  });

  it('renders correctly with selected option image', async () => {
    await act(async () => {
      rendered = render(
        <ThemeProvider theme={defaultTheme}>
          <Select
            options={testOptions}
            defaultSelectedId={testOptions[2].id}
            onChange={() => {}}
          />
        </ThemeProvider>
      );
    });

    expect(rendered.asFragment()).toMatchSnapshot();

    const select = rendered.container.children?.item(0) as Element;
    const selectedOption = select.children?.item(0) as Element;
    const selectedOptionLeft = selectedOption.children?.item(0) as Element;
    const selectedOptionImage = selectedOptionLeft.children?.item(0) as Element;
    expect(selectedOptionImage).toHaveAttribute('src', testOptions[2].imageSrc);
  });
});
