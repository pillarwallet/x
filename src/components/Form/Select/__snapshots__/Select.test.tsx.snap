// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly with default option 1`] = `
<DocumentFragment>
  .c4 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-right: 5px;
}

.c1 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c2 {
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

.c0:hover {
  background: rgba(217, 217, 217, 0.2);
}

<div>
    <div
      class="c0"
      id="action-bar-send-test 2"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 2
        </p>
        <p
          class="c3"
        >
          test2
        </p>
      </div>
      <div
        class="c4"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Select /> renders correctly with no options 1`] = `
<DocumentFragment>
  .c0 {
  padding: 13px 13px 13px;
  background: transparent;
  border: 1px solid rgba(226, 221, 255, 0.15);
  color: rgba(226, 221, 255, 0.9);
  margin-bottom: 15px;
  font-size: 12px;
  border-radius: 6px;
  min-height: 35px;
  display: flex;
  align-items: center;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  display: inline-block;
  margin-right: 10px;
}

<div>
    <p
      class="c0"
    >
      <span
        class="c1"
      >
        <svg
          fill="none"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 9v5M12 21.41H5.94c-3.47 0-4.92-2.48-3.24-5.51l3.12-5.62L8.76 5c1.78-3.21 4.7-3.21 6.48 0l2.94 5.29 3.12 5.62c1.68 3.03.22 5.51-3.24 5.51H12v-.01Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M11.995 17h.009"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </span>
      error.noNftsFound
    </p>
  </div>
</DocumentFragment>
`;

exports[`<Select /> renders correctly with option image 1`] = `
<DocumentFragment>
  .c5 {
  width: 42px;
  height: 42px;
  border-radius: 50%;
}

.c4 {
  width: 42px;
}

.c1 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c2 {
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

.c0:hover {
  background: rgba(217, 217, 217, 0.2);
}

<div>
    <div
      class="c0"
      id="action-bar-send-test 1"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 1
        </p>
        <p
          class="c3"
        >
          test1
        </p>
      </div>
    </div>
    <div
      class="c0"
      id="action-bar-send-test 2"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 2
        </p>
        <p
          class="c3"
        >
          test2
        </p>
      </div>
    </div>
    <div
      class="c0"
      id="action-bar-send-test 3"
    >
      <div
        class="c4"
      >
        <img
          alt="test 3"
          class="c5"
          src="https://via.placeholder.com/150"
          title="test 3"
        />
      </div>
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 3
        </p>
        <p
          class="c3"
        >
          test3
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Select /> renders correctly with selected option image 1`] = `
<DocumentFragment>
  .c2 {
  width: 42px;
  height: 42px;
  border-radius: 50%;
}

.c6 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-right: 5px;
}

.c1 {
  width: 42px;
}

.c3 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c4 {
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
}

.c5 {
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

.c0:hover {
  background: rgba(217, 217, 217, 0.2);
}

<div>
    <div
      class="c0"
      id="action-bar-send-test 3"
    >
      <div
        class="c1"
      >
        <img
          alt="test 3"
          class="c2"
          src="https://via.placeholder.com/150"
          title="test 3"
        />
      </div>
      <div
        class="c3"
      >
        <p
          class="c4"
        >
          test 3
        </p>
        <p
          class="c5"
        >
          test3
        </p>
      </div>
      <div
        class="c6"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Select /> renders correctly with some options 1`] = `
<DocumentFragment>
  .c5 {
  width: 42px;
  height: 42px;
  border-radius: 50%;
}

.c4 {
  width: 42px;
}

.c1 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c2 {
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

.c0:hover {
  background: rgba(217, 217, 217, 0.2);
}

<div>
    <div
      class="c0"
      id="action-bar-send-test 1"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 1
        </p>
        <p
          class="c3"
        >
          test1
        </p>
      </div>
    </div>
    <div
      class="c0"
      id="action-bar-send-test 2"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 2
        </p>
        <p
          class="c3"
        >
          test2
        </p>
      </div>
    </div>
    <div
      class="c0"
      id="action-bar-send-test 3"
    >
      <div
        class="c4"
      >
        <img
          alt="test 3"
          class="c5"
          src="https://via.placeholder.com/150"
          title="test 3"
        />
      </div>
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 3
        </p>
        <p
          class="c3"
        >
          test3
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Select /> responds once clicked 1`] = `
<DocumentFragment>
  .c4 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-right: 5px;
}

.c1 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c2 {
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

.c0:hover {
  background: rgba(217, 217, 217, 0.2);
}

<div>
    <div
      class="c0"
      id="action-bar-send-test 2"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 2
        </p>
        <p
          class="c3"
        >
          test2
        </p>
      </div>
      <div
        class="c4"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Select /> shows loading skeleton on option item where option value is loading 1`] = `
<DocumentFragment>
  .c4 {
  height: 15px;
  width: 30%;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 8px;
  margin-bottom: 0;
}

.c6 {
  width: 42px;
  height: 42px;
  border-radius: 50%;
}

.c5 {
  width: 42px;
}

.c1 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c2 {
  font-size: 15px;
  font-weight: 500;
  word-break: break-word;
}

.c3 {
  font-size: 14px;
  font-weight: 400;
  word-break: break-word;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

.c0:hover {
  background: rgba(217, 217, 217, 0.2);
}

<div>
    <div
      class="c0"
      id="action-bar-send-test 1"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 1
        </p>
        <p
          class="c3"
        >
          test1
        </p>
      </div>
    </div>
    <div
      class="c0"
      id="action-bar-send-test 2"
    >
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 2
        </p>
        <div
          class="c4"
        />
      </div>
    </div>
    <div
      class="c0"
      id="action-bar-send-test 3"
    >
      <div
        class="c5"
      >
        <img
          alt="test 3"
          class="c6"
          src="https://via.placeholder.com/150"
          title="test 3"
        />
      </div>
      <div
        class="c1"
      >
        <p
          class="c2"
        >
          test 3
        </p>
        <p
          class="c3"
        >
          test3
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Select /> shows loading skeleton when options loading 1`] = `
<DocumentFragment>
  .c2 {
  height: 42px;
  width: 42px;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 50%;
  margin-bottom: 0;
}

.c4 {
  height: 15px;
  width: 80%;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 8px;
  margin-bottom: 0;
}

.c5 {
  height: 15px;
  width: 30%;
  animation: hLhBjP 1s linear infinite alternate;
  border-radius: 8px;
  margin-bottom: 0;
}

.c1 {
  width: 42px;
}

.c3 {
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c0 {
  border-radius: 10px;
  background: rgba(217, 217, 217, 0.05);
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-content: center;
  gap: 12px;
  padding: 11px 13px;
  cursor: pointer;
  user-select: none;
  margin-bottom: 10px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      />
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      />
      <div
        class="c5"
      />
    </div>
  </div>
</DocumentFragment>
`;
